container-images/intel-gpu/Containerfile

```diff
@@ -0,0 +1,32 @@
+FROM quay.io/fedora/fedora:41 as builder
+
+COPY intel-gpu/oneAPI.repo /etc/yum.repos.d/
+
+RUN dnf install -y lspci clinfo intel-opencl g++ cmake git libcurl-devel intel-oneapi-base-toolkit ; \
+    git clone https://github.com/ggerganov/llama.cpp.git -b b4523 ; \
+    cd llama.cpp ; \
+    mkdir -p build ; \
+    cd build ; \
+    source /opt/intel/oneapi/setvars.sh ; \
+    cmake .. -DGGML_SYCL=ON -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx -DLLAMA_CURL=ON -DGGML_CCACHE=OFF -DGGML_NATIVE=ON ; \
+    cmake --build . --config Release -j -v ; \
+    cmake --install . --prefix /llama-cpp
+
+FROM quay.io/fedora/fedora:41
+
+ENV LD_LIBRARY_PATH="/usr/local/lib64:/usr/local/lib/:/opt/intel/oneapi/mkl/2025.0/lib:/opt/intel/oneapi/compiler/2025.0/opt/compiler/lib:/opt/intel/oneapi/compiler/2025.0/lib/clang/19/lib:/opt/intel/oneapi/compiler/2025.0/lib:/opt/intel/oneapi/umf/0.9/lib:/opt/intel/oneapi/tbb/2022.0/lib:/opt/intel/oneapi/tcm/1.2/lib:/opt/intel/oneapi/redist/opt/compiler/lib:/opt/intel/oneapi/redist/lib/clang/19/lib:/opt/intel/oneapi/redist/lib:/opt/intel/oneapi/mkl/2025.0/lib:/opt/intel/oneapi/compiler/2025.0/opt/compiler/lib:/opt/intel/oneapi/compiler/2025.0/lib/clang/19/lib:/opt/intel/oneapi/compiler/2025.0/lib:/opt/intel/oneapi/umf/0.9/lib:/opt/intel/oneapi/tbb/2022.0/lib:/opt/intel/oneapi/tcm/1.2/lib:/opt/intel/oneapi/redist/opt/compiler/lib:/opt/intel/oneapi/redist/lib/clang/19/lib:/opt/intel/oneapi/redist/lib"
+
+COPY --from=builder /llama-cpp/bin/ /usr/local/bin/
+COPY --from=builder /llama-cpp/lib/ /usr/local/lib/
+COPY --from=builder /llama-cpp/lib64/ /usr/local/lib64/
+COPY --from=builder /llama-cpp/include/ /usr/local/include/
+COPY intel-gpu/oneAPI.repo /etc/yum.repos.d/
+
+RUN dnf install -y intel-opencl libcurl lspci clinfo intel-oneapi-runtime-compilers intel-oneapi-mkl-core intel-oneapi-mkl-sycl-blas intel-oneapi-runtime-dnnl ; \
```

Comment:
To improve build time, I would suggest using additional "base" stage that install these libraries, and which is used both by build, and final container stages.

That way packages common to final and build stages would be installed only once.